File: examples/benchmarks/fib.rb
#! ruby
#
# this algorithm is so dumb!
# it even hurts to write this equivalent to fib.pl

def fib(n)
	return n if (n < 2)
	return fib(n - 1) + fib(n - 2)
end

N = Integer( ARGV.shift || 28 )

puts "fib(#{N}) = #{ fib(N) }"

File: examples/benchmarks/fib.py
#! python

import sys

def fib(n):
    if (n < 2):
        return(n)
    return( fib(n-2) + fib(n-1) )

N = int(len(sys.argv) == 2 and sys.argv[1] or 24)

print "fib(%d) = %d" %( N, fib(N) )
